tools/clang/unittests/HLSLExec/LongVectors.h

```diff
@@ -12,403 +12,351 @@
+                              TableParameterHandler &Handler);
+
+  template <typename DataTypeT, typename LongVectorOpTypeT>
+  void DispatchTestByVectorSize(LongVectorOpTypeT OpType,
+                                TableParameterHandler &Handler);
+
+  template <typename DataTypeT, typename LongVectorOpTypeT>
+  void TestBaseMethod(
+      LongVector::TestConfig<DataTypeT, LongVectorOpTypeT> &TestConfig,
+      size_t VectorSizeToTest);
+
+private:
+  dxc::DxcDllSupport m_Support;
+  bool m_Initialized = false;
+};
 
-  bool operator>(const HLSLBool_t &other) const { return val > other.val; }
+template <typename DataTypeT>
+void FillShaderBufferFromLongVectorData(std::vector<BYTE> &ShaderBuffer,
+                                        std::vector<DataTypeT> &TestData);
 
-  bool operator<=(const HLSLBool_t &other) const { return val <= other.val; }
+template <typename DataTypeT>
+void FillLongVectorDataFromShaderBuffer(MappedData &ShaderBuffer,
+                                        std::vector<DataTypeT> &TestData,
+                                        size_t NumElements);
 
-  bool operator>=(const HLSLBool_t &other) const { return val >= other.val; }
+template <typename DataTypeT> constexpr bool IsFloatingPointType() {
+  return std::is_same_v<DataTypeT, float> ||
+         std::is_same_v<DataTypeT, double> ||
+         std::is_same_v<DataTypeT, HLSLHalf_t>;
+}
 
-  HLSLBool_t operator*(const HLSLBool_t &other) const {
-    return HLSLBool_t(val * other.val);
-  }
+struct LongVectorOpTypeStringToEnumValue {
+  std::wstring OpTypeString;
+  uint32_t OpTypeValue;
+};
 
-  HLSLBool_t operator+(const HLSLBool_t &other) const {
-    return HLSLBool_t(val + other.val);
-  }
+template <typename DataTypeT>
+DataTypeT GetLongVectorOpType(const LongVectorOpTypeStringToEnumValue *Values,
+                              const std::wstring &OpTypeString,
+                              std::size_t Length);
 
-  // So we can construct std::wstrings using std::wostream
-  friend std::wostream &operator<<(std::wostream &os, const HLSLBool_t &obj) {
-    os << static_cast<bool>(obj.val);
-    return os;
-  }
+enum ValidationType {
+  ValidationType_Epsilon,
+  ValidationType_Ulp,
+};
 
-  // So we can construct std::strings using std::ostream
-  friend std::ostream &operator<<(std::ostream &os, const HLSLBool_t &obj) {
-    os << static_cast<bool>(obj.val);
-    return os;
-  }
+enum BasicOpType {
+  BasicOpType_Binary,
+  BasicOpType_Unary,
+  BasicOpType_ScalarBinary,
+  BasicOpType_EnumValueCount
+};
 
-  int32_t val = 0;
+enum BinaryOpType {
+  // Below this line to be moved to HLSLOpType (+, -, * etc)
+  BinaryOpType_ScalarAdd,
+  BinaryOpType_ScalarMultiply,
+  BinaryOpType_ScalarSubtract,
+  BinaryOpType_ScalarDivide,
+  BinaryOpType_ScalarModulus,
+  BinaryOpType_Multiply,
+  BinaryOpType_Add,
+  BinaryOpType_Subtract,
+  BinaryOpType_Divide,
+  BinaryOpType_Modulus,
+  // Below this line to be moved to HLSLMathOpType (min, max etc)
```

Comment:
I'm not sure what to make of these comments.  Is this a TODO?

Reply:
Yeah, essentially. I was making some notes while working should have probably just removed this.

In a subsequent PR when adding some of the math specific intrinsics I was considering adding a MathOpType_*.  And then they'd have a LongVector::BasicOpType::BasicOpType_Binary property set.

Reply:
Removing the comment and the one above it.